### ISSUE TITLE
Sharing violation on FAT32 drive

### ISSUE THREAD

**Issue by mrward** [mono/mono#19221]
Originally filed on Developer Community - https://developercommunity.visualstudio.com/content/problem/934824/files-on-an-exfat-drive-on-mac.html

Also possibly related - https://developercommunity.visualstudio.com/content/problem/528765/sharing-violation-on-fat-file-systems-on-osx-when.html

https://xamarin.github.io/bugzilla-archives/38/38538/bug.html

## Steps to Reproduce

1. dotnet new console
2. msbuild 

dotnet build works but Mono fails with:

CSC : error CS2012: Cannot open '/Volumes/FAT32Drive/test/obj/Debug/netcoreapp3.1/test.pdb' for writing -- 'Sharing violation on path /Volumes/FAT32Drive/test/obj/Debug/netcoreapp3.1/test.pdb' [/Volumes/NO NAME/test/test.csproj]

### Current Behavior

Sharing violation on path.

### Expected Behavior

Not sure if this is supported.

## On which platforms did you notice this

[x] macOS
[ ] Linux
[ ] Windows

**Version Used**:

Mono JIT compiler version 6.10.0.87 (2019-12/b33fa1d00ed Wed Feb 26 21:46:15 EST 2020)

**Comment by hamarb123** [comment]
Hey, please fix this.
I cannot do any work using my mac, and have to rely on virtualisation to get anything done, this is a really silly issue and shouldn't be too hard to fix I don't think.

**Comment by hamarb123** [comment]
@mrward 
It seems to be related to https://github.com/xamarin/xamarin-android/issues/3465. This type of project works on my exFAT (not FAT32 - please edit your description as per my original issue) drive.

**Comment by hamarb123** [comment]
Here is a copy from the log in Visual Studio:
Getting restore information for project /Volumes/Hamish 4TB/Hamish Arblaster/Software/Code/Tests/Youtube/SelfAwareVideoUploader/SelfAwareVideoUploader/SelfAwareVideoUploader.csproj
Sharing violation on path /Volumes/Hamish 4TB/Hamish Arblaster/Software/Code/Tests/Youtube/SelfAwareVideoUploader/SelfAwareVideoUploader/obj/SelfAwareVideoUploader.csproj.nuget.dgspec.json

And when you try to save an edited file, eventually you'll get:
<img width="600" alt="Screen Shot 2020-03-20 at 8 26 47 pm" src="https://user-images.githubusercontent.com/25316035/77150972-204c0a80-6ae9-11ea-82b8-6746c467af54.png">

FYI: that was just a test project file name

**Comment by hamarb123** [comment]
And https://github.com/mono/mono/issues/16032

**Comment by mrward** [comment]
@hamarb123 Not sure what you mean - it works on exFAT but not FAT? I tested on a FAT drive not an exFAT drive. What do you want the title to be changed to?

**Comment by hamarb123** [comment]
@mrward yes, I was just saying that it also doesn't work on exFAT drives, thanks for renaming it.

**Comment by thassin** [comment]
also see: https://github.com/mono/mono/issues/19461

**Comment by CMarcher** [comment]
Getting this when trying to build from a solution in a separate ExFAT volume on MacOS.

**Comment by Alexislml96** [comment]
I have the same issue, when i build a project in to an ExFat drive, the "Sharing Violation on Path" error appears.

**Comment by jgraygi** [comment]
Same here.  Works fine with .net core projects but anything with .net framework I get "Sharing Violation on Path" error.

**Comment by namquang93** [comment]
bump this for a possible workaround...
